gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/DefaultPubSubRpc.java

```diff
@@ -50,48 +54,78 @@
 import com.google.pubsub.v1.Subscription;
 import com.google.pubsub.v1.Topic;
 
+import io.grpc.ManagedChannel;
+import io.grpc.Status.Code;
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+
 import org.joda.time.Duration;
 
 import java.io.IOException;
 import java.util.Set;
 import java.util.concurrent.Future;
-
-import autovalue.shaded.com.google.common.common.collect.Sets;
-import io.grpc.Status.Code;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
 
 public class DefaultPubSubRpc implements PubSubRpc {
 
   private final PublisherApi publisherApi;
   private final SubscriberApi subscriberApi;
+  private final ScheduledExecutorService executor =
+      MoreExecutors.getExitingScheduledExecutorService(new ScheduledThreadPoolExecutor(8));
```

Comment:
> A comment?

Will add one. Also I believe `getExitingScheduledExecutorService` won't work on AE as it requires permission to register system hooks.

> Also, I don't think using MoreExecutors.getExitingScheduledExecutorService is a good idea unless that was static

Can you provide some more details? Why should local vs static declaration matter in this case?

> Regardless the static part, a better option would be to make sure we are using a thread pool that shrink
to zero threads after a reasonable period of no use.

The VTIK layer uses non-shrinkable pools so I guessed that was the preferred choice. /cc @garrettjonesgoogle 
Also we should decide the `corePoolSize` and `keepAliveTime` parameters, which I am not so sure about. I know that cached thread pool use `keepAliveTime=60seconds`. I wouldn't reduce `corePoolSize` to 0 though.